napari/components/viewer_model.py

```diff
@@ -607,6 +609,9 @@ def add_points(
         ----------
         data : array (N, D)
             Coordinates for N points in D dimensions.
+        annotations : dict
```

Comment:
(a) I would call this "dict {str: array (N,)}", and
(b) I would aim to support both dicts of columns and pandas dataframes, and, in the future, dask dataframes and cudf dataframes.

Reply:
Thanks for taking a look, @jni !

> (a) I would call this "dict {str: array (N,)}", and

Good point. I will fix this.

> (b) I would aim to support both dicts of columns and pandas dataframes, and, in the future, dask dataframes and cudf dataframes.

I agree with keeping compatibility with DataFrame and DataFrame-like things in mind for the annotations. There was some discussion in #858 about trying to avoid picking up dependencies, especially for the model, so I opted not to implement conversion methods here (for now). In terms of `dask` and `cudf` `DataFrame` support, I _think_ those might be dropin if we make our internal annotations datatype a `pandas` `DataFrame` (in a similar way, we can use dask arrays in the `Image` layer). I have tried to implement this such that we don't touch annotations too much and the slicing is simple, so if/when we decide to migrate to a `DataFrame`, I think it's a pretty straightforward refactor.

Since our annotations are fairly simple for now, I propose the following:

- We keep `Points.annotations` as a dictionary of arrays.
- When this lands, I will update the `Points` layer tutorial to include these new features. I will also add a section on converting `DataFrame` to our `Points.annotations` data type. That way if users already have annotations in a `DataFrame`, they can still take advantage of the annotations.
- If there is demand for native `DataFrame`, we can refactor and make `Points.annotations` a `DataFrame`. That might also be an opportunity to make sure the coordinates are also implemented in a way that can utilize dask arrays.

How does this sound?

Reply:
@kevinyamauchi I think we can support dataframes without depending on pandas, if you just get the data as `np.asarray(annotations[colname])`. This should work with annotations being a dict of arrays or a pandas dataframe.

Reply:
Oh cool, I didn't think of that! I'll try that out. Thanks, @jni!

If it works, I'd like to add a test. If I do, should I add `pandas` to /requirements/test.txt`?

Reply:
@jni, I just took this for a spin and it mostly works. The one wrinkle is adding and removing annotations (e.g., when adding or deleting a point). I am not super familiar with the pandas API, but I can't figure out a way to do so without the output being a dictionary. I think converting to a dictionary is fine if we want to simply support inputing a pandas dataframe, but I think we would lose the advantages of a dask or cudf DataFrame. 

For example, if we want to delete the the annotations at `index` (could be one index or a list of indices), we could use the following:

```python
self.annotations = {
    k: np.delete(v, index, axis=0) for k, v in self.annotations.items()
}
```

In pandas, I think we would usually use `df.drop()` to remove the whole row associated with the indices we would like to remove. We could modify the snippet above to accept a pandas DataFrame in `self.annotations`, but the output would be a dictionary. 

```python
self.annotations = {
    k: np.delete(np.asarray(self.annotations[k]), index, axis=0) \
        for k in self.annotations
}
```

Do you know of a better implementation? If not, perhaps a good middle ground is to convert if a DataFrame is provided, convert it into a dictionary when setting `Points.annotations`. We can then consider making `DataFrame` the internal representation later if there is demand. What do you think?

Reply:
Sorry, I forgot to respond here. I think conversion to our canonical form is the right move for now. The only question I have is whether it's necessary to directly convert to np.ndarray, rather than keeping the series in there.